crates/matrix-sdk/src/room/mod.rs

```diff
@@ -3702,18 +3697,6 @@ impl Room {
         opts.send(self, event_id).await
     }
 
-    /// Handle an [`AnySyncMessageLikeEvent`] in this room's [`RoomIndex`].
-    ///
-    /// This which will add/remove/edit an event in the index based on the
-    /// event type.
-    #[cfg(feature = "experimental-search")]
-    pub(crate) async fn index_event(
-        &self,
-        event: AnySyncMessageLikeEvent,
-    ) -> Result<(), IndexError> {
```

Comment:
I think you can remove the `EventCacheError::IndexError`, with all the changes I've proposed in this PR?